flytekit/image_spec/image_spec.py

```diff
@@ -44,7 +45,7 @@ class ImageSpec:
 
     name: str = "flytekit"
     python_version: str = None  # Use default python in the base image if None.
-    builder: str = "envd"
+    builder: Optional[str] = None
```

Comment:
won't this change force every existing client to specify "envd" in their uses of `ImageSpec`?

Reply:
If `"envd"` is not set and `flytekitplugins-envd` is installed, then `envd` is still used. As for testing, there is an existing test that does not have `builder` set and it still runs through `envd` :

https://github.com/flyteorg/flytekit/blob/139fa2fa1589bdc93ad1588d3b3e15fd8f55441a/plugins/flytekit-envd/tests/test_image_spec.py#L10-L16

With this PR, the default can switch when another image spec plugin is installed and it gives itself a higher priority. If a user still wants `envd` to be the default, they can reregister `envd` and give it a higher priority:

```python
from flytekitplugins.envd import EnvdImageSpecBuilder
from flytekit.image_spec.image_spec import ImageBuildEngine

ImageBuildEngine.register("envd", EnvdImageSpecBuilder(), priority=20)
```